### PR TITLE
Refactor AvailableJavaHomes to use new metadata format

### PR DESCRIPTION
**Pull request by bmuskalla**

### CODE REVIEW

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Comment by bmuskalla** [comment]
@bot-gradle test this

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AvailableJavaHomes.java [hunk 1]
```diff
@@ -116,25 +115,26 @@ public static Jvm getJdk(final JavaVersion version) {
      */
     public static List<Jvm> getJdks(JavaVersion... versions) {
         final Set<JavaVersion> remaining = Sets.newHashSet(versions);
-        return getAvailableJdks(element -> remaining.remove(element.getJavaVersion()));
+        return getAvailableJdks(element -> remaining.remove(element.getLangageVersion()));
```
Comment (ljacomet) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AvailableJavaHomes.java:118]: Totally missed that in other reviews - we have a typo there: `getLanguageVersion`

**Review by ljacomet — APPROVED** [review]
Changes all good. Would be nice to fix the typo, see below

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Pull request merged**